### ISSUE TITLE
unable to change the parent of modal using parentSelector props

### ISSUE THREAD

**Issue by icristal** [reactjs/react-modal#626]
I am unable to change the parent node of the modal even though parentSelector accepts a function. 

Take for example this code:

```
class Example extends React.Component {
  getParentNode () {
    return document.querySelector(this.props.anchorSelector);
  }

  render () {
    <ReactModal isOpen={this.props.isLoading} parentSelector={this.getParentNode.bind(this)}>
      <p>Something here</p>
    </ReactModal>
  }
}
```

so if my initial `anchorSelector` has a value of `div.initial-container` the modal opens with no problem. however if I close the modal and change the anchorSelector to `div.second-container` it does not use the second-container at all and still uses the initial one.

**Comment by diasbruno** [comment]
Merged MR #630, so now it should reparent the modal unconditionally.

**Comment by diasbruno** [comment]
I'll close this issue as it's resolved now. Please feel free to reopen if it's still an issue. 

Thanks, @icristal.
